mango-cli/src/main/scala/org/bdgenomics/mango/cli/VizReads.scala

```diff
@@ -220,54 +189,34 @@ class VizServlet extends ScalatraServlet {
-        } case None => VizReads.errors.outOfBounds
+          val d = VizReads.readsData.getFrequency(region, sampleIds)
+          Ok(d)
+        }
+        case None => VizReads.errors.outOfBounds
       }
     }
   }
 
-  get("/mergedReads/:ref") {
-    VizTimers.AlignmentRequest.time {
-      contentType = "json"
-      val dictOpt = VizReads.globalDict(params("ref"))
-      val viewRegion = ReferenceRegion(params("ref"), params("start").toLong, VizUtils.getEnd(params("end").toLong, dictOpt))
-      dictOpt match {
-        case Some(_) => {
-          val sampleIds: List[String] = params("sample").split(",").toList
-          val diffReads = params.getOrElse("diff", "0") != "0"
-          val dataOption = VizReads.readsData.multiget(viewRegion, sampleIds)
-          dataOption match {
-            case Some(_) => {
-              val binSize = VizUtils.getBinSize(viewRegion, VizReads.screenSize)
-              var alignmentData: Map[String, List[MutationCount]] = MergedAlignmentRecordLayout(dataOption.get.toRDD, binSize)
-              val fileMap = VizReads.readsData.getFileMap
-              var readRetJson: String = ""
-              if (diffReads) {
-                alignmentData = MergedAlignmentRecordLayout.diffRecords(sampleIds, alignmentData)
-              }
-              for (sample <- sampleIds) {
-                val sampleData = alignmentData.get(sample)
-                sampleData match {
-                  case Some(_) =>
-                    readRetJson += "\"" + sample + "\":" +
-                      "{ \"filename\": " + write(fileMap(sample)) +
-                      ", \"indels\": " + write(sampleData.get.filter(_.op != "M")) +
-                      ", \"mismatches\": " + write(sampleData.get.filter(_.op == "M")) +
-                      ", \"binSize\": " + binSize + "},"
-                  case None =>
-                    readRetJson += "\"" + sample + "\":" +
-                      "{ \"filename\": " + write(fileMap(sample)) + "},"
-                }
-
-              }
-              readRetJson = readRetJson.dropRight(1)
-              readRetJson = "{" + readRetJson + "}"
-              readRetJson
-            } case None => {
-              write("")
-            }
+  get("/reads/:ref") {
+    VizTimers.ReadsRequest.time {
+      val viewRegion = ReferenceRegion(params("ref"), params("start").toLong, params("end").toLong)
+      if (viewRegion.length() > 5000) {
+        VizReads.errors.largeRegion
+      } else {
+        val isRaw =
+          try {
+            params("isRaw").toBoolean
+          } catch {
+            case e: Exception => false
           }
-        } case None => VizReads.errors.outOfBounds
+        contentType = "json"
+        val dictOpt = VizReads.globalDict(viewRegion.referenceName)
+        // if region is in bounds of reference, return data
+        if (dictOpt.isDefined && viewRegion.end <= dictOpt.get.length) {
```

Comment:
because we get the globalDict from the reference file, not the reads file, couldn't it be the case that the information for the reads doesn't exist?